### PR TITLE
[serverless] fix missing report log spindown metrics

### PR DESCRIPTION
**Pull request by thedavl**
### What does this PR do?

Fixes a bug where we won't receive report log generated enhanced metrics when a lambda is idle and receives a spindown event. This happens with specific runtimes, I've noticed it mainly with low memory nodejs custom runtime container lambdas as well as ruby non-container lambdas, predominantly with 128 or 256mb.

AWS advertises we will have ~1.7 seconds after receiving a spindown event to do work, though in these specific runtimes it will sometimes be <1.3 seconds (AWS gives us the deadline timestamp alongside the spindown event). This causes the lambda to be shutdown before we are able to send the outgoing http request (as serialization and goroutine context switching takes hundreds of ms)

The fix for this is lowering the shutdown delay. We can do this as I've benchmarked how long it actually takes to receive and process the `REPORT` log from the lambda API, and it's fully processed on average just single milliseconds after getting the spindown event, sometimes even before that. Occasionally it will creep closer to 100ms.

Through RCs and self-monitoring apps, 200ms shutdown delay seems to be the sweetspot for very consistent reporting of report log metrics.

### Describe how to test/QA your changes

image
- Self-monitoring app of node custom runtime container, node alpine custom runtime container, and python custom runtime container lambdas
image
- No regressions in our old deprecated self-monitoring app
image
- No regressions in our new RC self-monitoring app
image
- new RC self-monitoring app cold start lambdas are stable before & after the changes

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/serverless/daemon/daemon.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ import (
 // ShutdownDelay is the amount of time we wait before shutting down the HTTP server
 // after we receive a Shutdown event. This allows time for the final log messages
 // to arrive from the Logs API.
-var ShutdownDelay time.Duration = 1 * time.Second
+var ShutdownDelay = 200 * time.Millisecond
```
Comment (IvanTopolcic) [inline pkg/serverless/daemon/daemon.go:30]: Just confirming, in the description you mentioned 400ms is the sweet spot. Why 200 here?

Comment (thedavl) [reply pkg/serverless/daemon/daemon.go:30]: ah typo, I meant 200. All the screenshots of self-monitoring & RCs are with 200

Comment (IvanTopolcic) [reply pkg/serverless/daemon/daemon.go:30]: Gotcha. And are we more likely to drop logs with this PR or is that something that you haven't noticed?

Comment (thedavl) [reply pkg/serverless/daemon/daemon.go:30]: Haven't noticed any dropped logs through self-monitoring, I benchmarked how long it actually takes to receive the report log, and 90+% of the time it's within single milliseconds.

**Review by IvanTopolcic — APPROVED** [review]

**Pull request merged**